PIL/GifImagePlugin.py

```diff
@@ -602,79 +615,64 @@ def _get_palette_bytes(im, palette, info):
         else:
             source_palette = bytearray([i//3 for i in range(768)])
 
-    used_palette_colors = palette_bytes = None
+    palette_bytes = None
 
-    if _get_optimize(im, info):
-        used_palette_colors = _get_used_palette_colors(im)
+    used_palette_colors = _get_optimize(im, info)
```

Comment:
This line is horribly confusing. `_get_optimize()` does not sound like it should return either a palette modification map or None. And there's no documentation to that effect.